check/src/typecheck.rs

```diff
@@ -727,9 +741,9 @@ impl<'a> Typecheck<'a> {
                 let mut match_type = self.remove_alias(match_type);
                 let mut types = Vec::new();
                 let new_type = match *match_type {
```

Comment:
For a reason have I have not been able to figure out (yet) I got a type error from the use of `make_Applicative` in the prelude. Added a minimized version of the issue to `pass.rs` but I have not been able to figure out a solution yet. Without the simplification it still works however.